### ISSUE TITLE
(bug) false positive when landmarks of different types are nested

### ISSUE THREAD

**Issue by zelaznik** [ember-template-lint/ember-template-lint#1542]
The following code raises an error with the role `no-nested-landmark`.

```hbs
<footer>
  <nav>
  </nav>
</footer>
```

I'm attaching a pull request that I believe fixes the problem:

https://github.com/ember-template-lint/ember-template-lint/pull/1541
